app/models/draft_vaccination_record.rb

```diff
@@ -367,4 +368,22 @@ def delivery_site_matches_delivery_method
       errors.add(:delivery_site, :inclusion)
     end
   end
+
+  def patient_attended_session
+    if session&.requires_registration? &&
+         !PatientSession::RegistrationStatus
+           .for_patient_session(patient, session)
+           .allows_recording_vaccinations
+           .exists?
+      date = performed_at.to_date
+      errors.add(
+        :session_id,
+        :patient_not_attending,
+        patient_full_name: patient.full_name,
+        display_date:
+          date.today? ? "Today (#{date.to_fs(:long)})" : date.to_fs(:long),
+        location_name:
```

Comment:
`location_name` is an optional value that's only set when recording a vaccination in a clinic session. I think you'll want to use the logic from [the `vaccination_record_location` helper](https://github.com/nhsuk/manage-vaccinations-in-schools/blob/8b814db86a61f49259c9a2ec99f5ac64f9bef3f0/app/helpers/vaccination_records_helper.rb#L4).